internal/test/e2e/run.go

```diff
@@ -482,3 +470,47 @@ func logTestGroups(logger logr.Logger, instancesConf []instanceRunConf) {
 	}
 	logger.V(1).Info("Running tests in parallel", "testsGroups", testGroups)
 }
+
+func getHardwarePool(storageBucket string) ([]*api.Hardware, error) {
+	awsSession, err := session.NewSession()
+	if err != nil {
+		return nil, fmt.Errorf("creating aws session for tests: %v", err)
+	}
+	err = s3.DownloadToDisk(awsSession, os.Getenv(tinkerbellHardwareS3FileKeyEnvVar), storageBucket, e2eHardwareCsvFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download tinkerbell hardware csv: %v", err)
+	}
+
+	hardware, err := api.NewHardwareSliceFromFile(e2eHardwareCsvFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Tinkerbell hardware: %v", err)
+	}
+	return hardware, nil
+}
+
+func reserveTinkHw(conf instanceRunConf, invQueue *hwQueue) {
+	now := time.Now()
+	after := now.Add(hwPollingTimeout)
+	for {
+		if now.After(after) {
+			conf.logger.V(1).Info("TestRunner timed out waiting for hardware", "instanceId", conf.instanceId)
+			break
+		}
+		invQueue.mu.Lock()
+		if conf.hardwareCount <= len(invQueue.hws) {
+			conf.logger.V(1).Info("Reserved hardware for TestRunner", "instanceId", conf.instanceId, "hardwarePool", conf.hardware)
+			conf.hardware = invQueue.dequeueHw(conf.hardwareCount)
```

Comment:
Let's re-think on how we want to handle the hardware reservation. I don't think handling enqueuing and dequeuing hardware at the test level is the best idea. If the test thread crashes or fails for some reason, that the hardware it reserved is lost.
I think what you should do from the tests' perspective is to just ask the hardware reservation system to reserve some number of hardware. The hardware reservation system should handle the implementation details on how to it wants to reserve the hardware, for example enqueuing and de-queuing the hardware. Also, there when you dequeue the hardware, you should store a list of hardware that is reserved by a test in a map so that info is not lost.